### PR TITLE
Add tutorial showing how to run CRI-O in kind

### PR DESCRIPTION
**Pull request by roman-kiselenko**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Small tutorial: How to run CRI-O as container runtime in kind.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by roman-kiselenko** [comment]
@cri-o/cri-o-maintainers PTAL 🙏

**Comment by haircommander** [comment]
thanks! you've got some markdown lint failures

File: tutorial.md [hunk 1]
```diff
@@ -8,3 +8,5 @@
 - To enable support for image decryption, see our [Image decryption support guide](tutorials/decryption.md).
 - For instructions on how to use CRI-O's experimental userns annotation, see the
   [documentation](tutorials/userns.md).
+- Learn how to build a KIND (Kubernetes IN Docker) node image with a specific Kubernetes version and CRI-O container runtime.
+  [CRI-O in KIND](tutorials/crio-in-kind.md).
```
Comment (kwilczynski) [inline tutorial.md:12]:
```suggestion
- Learn how to build a kind (Kubernetes IN Docker) node image with a specific Kubernetes version and CRI-O container runtime.
  [CRI-O in KIND](tutorials/crio-in-kind.md).
```

File: tutorials/crio-in-kind.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
+# CRI-O in KIND
```
Comment (kwilczynski) [inline tutorials/crio-in-kind.md:1]:
```suggestion
# CRI-O in kind
```

File: tutorials/crio-in-kind.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+This tutorial will walk you through the creation of [KIND](https://kind.sigs.k8s.io/)
+cluster with CRI-O Container Runtime by creating a custom [`node image`](https://kind.sigs.k8s.io/docs/design/node-image/).
+
+It assumes you've already installed `golang>=1.22.2`,`kind>=v0.22.0`,`docker>=26.1.0`.
+
+ - [Build Base Image](#build-base-image)
+ - [Build Node Image](#build-node-image)
+ - [Create KIND cluster](#create-kind-cluster)
```
Comment (kwilczynski) [inline tutorials/crio-in-kind.md:10]:
```suggestion
 - [Create kind cluster](#create-kind-cluster)
```

File: tutorials/crio-in-kind.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      criSocket: unix:///var/run/crio/crio.sock
+- role: worker
+  kubeadmConfigPatches:
+  - |
+    kind: JoinConfiguration
+    nodeRegistration:
+      criSocket: unix:///var/run/crio/crio.sock
+```
+
+### Create KIND cluster
```
Comment (kwilczynski) [inline tutorials/crio-in-kind.md:128]:
```suggestion
### Create kind cluster
```

File: tutorials/crio-in-kind.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
+# CRI-O in KIND
+
+This tutorial will walk you through the creation of [KIND](https://kind.sigs.k8s.io/)
```
Comment (kwilczynski) [inline tutorials/crio-in-kind.md:3]:
```suggestion
This tutorial will walk you through the creation of [kind](https://kind.sigs.k8s.io/)
```

**Comment by kwilczynski** [comment]
@roman-kiselenko, given that it's based on your excellent blog post, feel free to ignore my suggestions for the changes to the way how "kind" is styled.

**Review by maiqueb — COMMENTED** [review]
Thanks for this. Amazing. Managed to create the cluster with the provided information.

One quick remark though; I have faced an issue with my configuration (I had docker configured to use buildx ...). I could not build the crio image if using buildx ... 

Not sure if you want to highlight that. 

Thanks again.

**Comment by mrunalp** [comment]
Awesome! We could keep this updated for newer versions on main branch and release branches will retain version specific instructions.

File: tutorials/crio-in-kind.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+      criSocket: unix:///var/run/crio/crio.sock
+```
+
+## Create kind cluster
+
+Let's create a cluster:
+
+<!-- markdownlint-disable MD013 -->
+```sh
+$ kind create cluster --image kindnode/crio:$CRIO_VERSION --config ./kind-crio.yaml
```
Comment (mrunalp) [inline tutorials/crio-in-kind.md:151]: Is it possible to use podman for running the cluster as an alternative to docker?

**Comment by kwilczynski** [comment]
> One quick remark though; I have faced an issue with my configuration (I had docker configured to use buildx ...). I could not build the crio image if using build ...

@maiqueb, what sort of issue shave you had? Any errors you can share we could have a look at?

This might be something we want to fix, potentially.

File: tutorials/crio-in-kind.md (hunk 6 above)
Comment (maiqueb) [reply tutorials/crio-in-kind.md:151]: AFAIK yes ... BUT generating the node and base images requires docker. Or used to require :) 

Probably better if someone confirmed this.

**Comment by maiqueb** [comment]
> > One quick remark though; I have faced an issue with my configuration (I had docker configured to use buildx ...). I could not build the crio image if using build ...
> 
> @maiqueb, what sort of issue shave you had? Any errors you can share we could have a look at?
> 
> This might be something we want to fix, potentially.

Sure thing @kwilczynski. The error happens when I try to build the CRIO img from the node image:
```bash
➜  new-cni-verbs-demo docker build --build-arg CRIO_VERSION=$CRIO_VERSION -t kindnode/crio:$CRIO_VERSION .
[+] Building 1.4s (2/2) FINISHED                                                                                    docker-container:kind-builder
 => [internal] load build definition from Dockerfile                                                                                         0.0s
 => => transferring dockerfile: 977B                                                                                                         0.0s
 => ERROR [internal] load metadata for docker.io/kindest/node:latest                                                                         1.2s
------
 > [internal] load metadata for docker.io/kindest/node:latest:
------
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
Dockerfile:1
--------------------
   1 | >>> FROM kindest/node:latest
   2 |     
   3 |     ARG CRIO_VERSION
--------------------
ERROR: failed to solve: kindest/node:latest: failed to resolve source metadata for docker.io/kindest/node:latest: docker.io/kindest/node:latest: not found

# I'm using buildx
➜  new-cni-verbs-demo cat ~/.docker/config.json      
{
	"auths": {},
	"aliases": {
		"builder": "buildx"
	}
}# 
```

The weird thing IMO is the image is actually available in `docker images`: 
```bash
➜  new-cni-verbs-demo docker images
REPOSITORY                                  TAG                                              IMAGE ID       CREATED         SIZE
...
kindest/node                                latest                                           7acfe6ceb949   28 hours ago    1.06GB
...
```

This was "fixed" by ... not using buildx :) 
```bash
➜  new-cni-verbs-demo mv ~/.docker/config.json ~/.docker/config.jsonBACKUP 
➜  new-cni-verbs-demo docker build --build-arg CRIO_VERSION=$CRIO_VERSION -t kindnode/crio:$CRIO_VERSION .
[+] Building 0.3s (6/6) FINISHED                                                                                                   docker:default
 => [internal] load build definition from Dockerfile                                                                                         0.0s
 => => transferring dockerfile: 977B                                                                                                         0.0s
 => [internal] load metadata for docker.io/kindest/node:latest                                                                               0.0s
 => [internal] load .dockerignore                                                                                                            0.0s
 => => transferring context: 2B                                                                                                              0.0s
 => [1/2] FROM docker.io/kindest/node:latest                                                                                                 0.0s
 => CACHED [2/2] RUN echo "Installing Packages ..."     && apt-get clean     && apt-get update -y     && DEBIAN_FRONTEND=noninteractive apt  0.0s
 => exporting to image                                                                                                                       0.0s
 => => exporting layers                                                                                                                      0.0s
 => => writing image sha256:fe153b4a09c05031acc6dc4dc09ec5563df58f5d9332320b17a859918b0caf84                                                 0.0s
 => => naming to docker.io/kindnode/crio:v1.30              
```

I've also tried to use `--load` when building, to no avail:
```bash
➜  new-cni-verbs-demo mv ~/.docker/config.jsonBACKUP ~/.docker/config.json 
➜  new-cni-verbs-demo docker build --build-arg CRIO_VERSION=$CRIO_VERSION --load -t kindnode/crio:$CRIO_VERSION .
[+] Building 0.6s (2/2) FINISHED                                                                                    docker-container:kind-builder
 => [internal] load build definition from Dockerfile                                                                                         0.0s
 => => transferring dockerfile: 977B                                                                                                         0.0s
 => ERROR [internal] load metadata for docker.io/kindest/node:latest                                                                         0.4s
------
 > [internal] load metadata for docker.io/kindest/node:latest:
------
Dockerfile:1
--------------------
   1 | >>> FROM kindest/node:latest
   2 |     
   3 |     ARG CRIO_VERSION
--------------------
ERROR: failed to solve: kindest/node:latest: failed to resolve source metadata for docker.io/kindest/node:latest: docker.io/kindest/node:latest: not found
```

File: tutorials/crio-in-kind.md (hunk 6 above)
Comment (roman-kiselenko) [reply tutorials/crio-in-kind.md:194]: Yeah, AFAIK, there is a lot of different options, and by leveraging `KIND_EXPERIMENTAL_PROVIDER` variable, you can create a cluster with different runtimes:

- podman
- nerdctl.lima
- finch
- nerdctl

**Review by maiqueb — APPROVED** [review]

**Comment by haircommander** [comment]
/approve

LGTM, thank you!
@cri-o/cri-o-maintainers PTAL

**Pull request merged**